tools/pubsys/src/vmware/govc.rs

```diff
@@ -0,0 +1,178 @@
+//! run specific `govc` commands.
+use duct::cmd;
+use log::trace;
+use pubsys_config::vmware::{Datacenter, DatacenterCreds};
+use snafu::ResultExt;
+use std::env;
+use std::path::Path;
+use std::process::Output;
+
+pub(crate) struct Govc {
+    env_config: Vec<String>,
+}
+
+impl Govc {
+    const GOVC: &'static str = "/usr/bin/govc";
+
+    /// Make a new instance of `Govc`, creating all of the environment variables required to run
+    /// `govc` as Docker `--env` arguments
+    pub(crate) fn new(dc: Datacenter, creds: DatacenterCreds) -> Self {
+        let mut env_config = Vec::new();
+        env_config.env_arg("GOVC_USERNAME", creds.username);
+        env_config.env_arg("GOVC_PASSWORD", creds.password);
+        env_config.env_arg("GOVC_URL", dc.vsphere_url);
+        env_config.env_arg("GOVC_DATACENTER", dc.datacenter);
+        env_config.env_arg("GOVC_DATASTORE", dc.datastore);
+        env_config.env_arg("GOVC_NETWORK", dc.network);
+        env_config.env_arg("GOVC_RESOURCE_POOL", dc.resource_pool);
+        env_config.env_arg("GOVC_FOLDER", dc.folder);
+
+        Self { env_config }
+    }
+
+    /// Run `govc import.ova` using Docker.
+    ///
+    /// Using the given name, OVA path, and import spec path, this function builds the `govc
+    /// import.ova` command as it will be used in the container.  It also builds the necessary bind
+    /// mount arguments to mount the import spec and OVA into the container.  Finally, it calls
+    /// `govc` via `docker run` invocation using these arguments.
+    pub(crate) fn upload_ova<S, P1, P2>(
+        self,
+        name: S,
+        ova_path: P1,
+        import_spec_path: P2,
+    ) -> Result<Output>
+    where
+        S: AsRef<str>,
+        P1: AsRef<Path>,
+        P2: AsRef<Path>,
+    {
+        let name = name.as_ref();
+        let ova_host_path = ova_path.as_ref();
+        let import_spec_host_path = import_spec_path.as_ref();
+
+        // Define the paths to the OVA and import spec we will use for the bind mounts into the
+        // container
+        let ova_container_path = "/tmp/bottlerocket.ova";
+        let import_spec_container_path = "/tmp/import.spec";
+
+        //--mount type=bind,source="path/to/thing",target=/tmp/thing,readonly
+        let mount_config = &[
+            // Mount the import spec file
+            "--mount",
+            &format!(
+                "type=bind,source={},target={},readonly",
+                import_spec_host_path.display(),
+                import_spec_container_path
+            ),
+            // Mount the OVA
+            "--mount",
+            &format!(
+                "type=bind,source={},target={},readonly",
+                ova_host_path.display(),
+                ova_container_path
+            ),
+        ];
+
+        // govc import.ova -options=/path/to/spec -name bottlerocket_vm_name /path/to/ova
+        let govc_cmd = &[
+            Self::GOVC,
```

Comment:
I think it's too strict to require that govc be located at `/usr/bin/govc`. I'm thinking you should just call `govc` and it needs to be in the `PATH`. That could be augmented with a parameter that allows you to specify the path to govc.

Regardless, I think the most important thing is that the error message be super clear that govc is a required binary and that we didn't find it where we expected it.

I was actually thinking that a check should be performed first so that a custom error message can be printed when govc isn't being found. As it is right now I think you might get something generic that could be confusing.

Reply:
Good point - I think we should just assume that `govc` is in the path.  The error message we'll get is pretty clear (I got it initially when I was testing this with a container that didn't have `govc` :) )

Reply:
I had forgotten that we were running it via the SDK. But the change to expect it in PATH still makes sense.